app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,14 @@
+class PiecesController < ApplicationController
+  def update
+    @piece = Pieces.find(params[:id])
+    new_x = params[:x].to_i
+    new_y = params[:y].to_i
+    return @piece.move!(new_x, new_y)
```

Comment:
You'll want to remove this `return` here actually (I believe I originally had it) since it's redundant (unless there's another return, the last line of a method in Ruby is returned)